### PR TITLE
remove personalisation from db

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

We previously removed the recipient's phone number from the notifications table as part of trying to avoid our limitation of not having more than 10000 rows of data in notifications that contains PII.  However, the phone number is present still in that table in another form -- encrypting inside the personalisation column, which contains everything in the csv row for that notification.

So the goal of this PR is to stop storing this personalisation info in the notifications table, and instead when we need personalisation info, we will fetch it from the ExpiringDict (our little cache) and if it is not present there, then we will fetch it from S3.

At this time we have not actually removed the phone_number and personalisation columns from the notifications table, we just no longer write to them.  To remove these columns breaks many tests and requires many code changes and at the time this PR was made, there was another PR in progress (the "enums" PR) which was making similar, but conflicting changes everywhere.

## Security Considerations

While not a security consideration per se, the goal of this PR is to move PII out of the database and rely on fetching it from S3.

### CODE REVIEW

**Comment by ccostino** [comment]
@terrazoon, could you please add a full description in here so that we can prep this PR for peer review?  You can use the [pull request template](https://github.com/GSA/notifications-api/blob/main/.github/pull_request_template.md) if you need a reference of what to include.

Once that information is in there, we can make sure it gets a peer review. 🙁 Thanks!

File: tests/app/celery/test_service_callback_tasks.py [hunk 1]
```diff
@@ -60,10 +60,15 @@ def test_send_delivery_status_to_service_post_https_request_to_service_with_encr
         "template_version": 1,
     }
 
+    # TODO why is 'completed_at' showing real time unlike everything else and does it matter?
```
Comment (A-Shumway42) [inline tests/app/celery/test_service_callback_tasks.py:63]: Does this mean `completed_at` isn't coming back as a string like `created_at` and `sent_at`? If so...I'm not sure about this one.

**Review by A-Shumway42 — APPROVED** [review]
It looks great to me. I don't have major questions but if anyone takes a look and catches something I'm all for understanding.

File: tests/app/celery/test_service_callback_tasks.py (hunk 1 above)
Comment (terrazoon) [reply tests/app/celery/test_service_callback_tasks.py:63]: We don't use 'completed_at' for anything as far as I know and we currently have the same problem with 'updated_at' that broke about 6-8 weeks ago.   So I will write a ticket to investigate those items, but I don't think this issue blocks the PR.

File: tests/app/user/test_rest_verify.py [hunk 2]
```diff
@@ -510,8 +514,12 @@ def test_send_user_email_code_with_urlencoded_next_param(
         _data=data,
         _expected_status=204,
     )
-    noti = Notification.query.one()
-    assert noti.personalisation["url"].endswith("?next=%2Fservices")
+    # TODO We are stripping out the personalisation from the db
+    # It should be recovered -- if needed -- from s3, but
+    # the purpose of this functionality is not clear.  Is this
+    # 2fa codes for email users?  Sms users receive 2fa codes via sms
```
[the remark this answers is not in the document]
Comment (ccostino) [reply tests/app/user/test_rest_verify.py:520]: Thanks for catching that!

**Review by ccostino — APPROVED** [review]
Thank you, @terrazoon!

**Pull request merged**